src/karts/abstract_kart.cpp

```diff
@@ -79,6 +99,12 @@ void AbstractKart::reset()
 }   // reset
 
 // ----------------------------------------------------------------------------
+const AbstractCharacteristic* AbstractKart::getCharacteristic() const
+{
+    return &(*m_characteristic);
```

Comment:
I can't use unique_ptr here because it wouldn't be unique then anymore. The safe way would be a weak_ptr, but you would need a shared_ptr for that and there would be overhead at runtime and in the code (because you need to cast a weak_ptr before you use it and so on...).
I decided to use a raw pointer here because it's most understandable. The only thing that has to be guaranteed for this to work, is that the pointer isn't used anymore when the parent object is deleted.